src/app/charts/line-graph.component.ts

```diff
@@ -279,26 +296,106 @@ export class LineGraphComponent {
         }
     }
 
+    private drawScrubber(): void {
+        // Vertical scrub line. Exists outside scrubber cluster because it moves independently
+        this.svg.append('line')
+            .attr('class', 'scrubline' + ' ' + this.id)
+            .attr('x1', 0).attr('x2', 0)
+            .attr('y1', 0).attr('y2', this.height)
+            .classed('hidden', true);
+
+        // Other scrubber elements
+        this.scrubber = this.svg.append('g')
+            .attr('class', this.id)
+            .classed('hidden', true);
+
+        this.scrubber.append('circle')
+            .attr('r', 4.5);
+
+        this.scrubber.append('rect')
+            .attr('class', 'scrubber-box' + ' ' + this.id)
+            .attr('height', 20);
+
+        this.scrubber.append('text')
+            .attr('class', 'scrubber-text' + ' ' + this.id);
+
+        // Scrubber sensory zone (set to size of graph) intentionally drawn last
+        // It overlays all other svg elements for uncompromised mouseover detection
+        this.svg.append('rect')
+            .attr('id', 'overlay')
+            .attr('height', this.height)
+            .attr('width', this.width);
+
+        // Toggle scrubber visibility
+        this.hover? $('.' + this.id).toggleClass('hidden', false) : $('.' + this.id).toggleClass('hidden', true);
+    }
+
+    private redrawScrubber(event) {
+        let xPos = event.offsetX - this.margin.left;
+        // Firefox handles event positioning differently than Chrome, Safari
+        if (navigator.userAgent.indexOf('Firefox') != -1) {
```

Comment:
Might want to change the check to `if (navigator && navigator.userAgent.toLowerCase().indexOf('Firefox') !== -1)`. These changes allow:
- The script to not throw an exception if the navigator global is missing (i.e. in tests)
- A case insensitive check on the UA string (which is a bit less specific)
- Triple-equals which is more correct and gets you into the habit of using triple equals by default for exact type equality